services/storage/src/simcore_service_storage/s3_client.py

```diff
@@ -303,16 +275,21 @@ async def delete_files_in_path(self, bucket: S3BucketName, *, prefix: str) -> No
 
         # NOTE: deletion of objects is done in batches of max 1000 elements,
         # the maximum accepted by the S3 API
-        async for s3_objects in self.list_all_objects_gen(
-            bucket,
-            prefix=prefix,
-            max_yield_result_size=_DELETE_OBJECTS_MAX_ACCEPTED_ELEMENTS,
-        ):
-            if objects_to_delete := [f["Key"] for f in s3_objects if "Key" in f]:
-                await self.client.delete_objects(
-                    Bucket=bucket,
-                    Delete={"Objects": [{"Key": key} for key in objects_to_delete]},
-                )
+        with log_context(_logger, logging.INFO, "deleting objects", log_duration=True):
+            delete_tasks: list[asyncio.Task] = []
+            async for s3_objects in self.list_all_objects_gen(bucket, prefix=prefix):
+                if objects_to_delete := [f["Key"] for f in s3_objects if "Key" in f]:
+                    delete_task = asyncio.create_task(
+                        self.client.delete_objects(
+                            Bucket=bucket,
+                            Delete={
+                                "Objects": [{"Key": key} for key in objects_to_delete]
+                            },
+                        )
+                    )
+                    delete_tasks.append(delete_task)
+
+            await logged_gather(*delete_tasks)
```

Comment:
I am not sure I see the point, now instead of keeping all the million objects in one data structure, now you keep them in several. but all in all the memory footprint is the same no? instead of accumulating all the objects in one list, you put them in separate ones.
The idea of the generator was to take 1000, then delete, then take the next 1000, then delete...
I guess you could define some upper limit than is more than 1000 but having it completely unbound looks a bit dangerous. Any comment? did you check the memory footprint vs number of objects?

Reply:
I measured it and apparently the memory footprint for 10k files is 10MB.

Reply:
@sanderegg moved this to serial deletion as agreed and will delete batches of 500 files not 1k. Reduces memory pressure.
Performance is similar between deleting 1000 files or 500 files at a time